src/Microsoft.Azure.WebJobs.Host/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -0,0 +1,460 @@
+
+        public BlobLeaseDistributedLockManager(
+            IStorageAccountProvider accountProvider,
+            IWebJobsExceptionHandler exceptionHandler, 
+            TraceWriter trace,
+            ILogger logger)
+        {
+            _accountProvider = accountProvider;
+            _exceptionHandler = exceptionHandler;
+            _trace = trace;
+            _logger = logger;
+        }
+
+        // for testing
+        internal TimeSpan MinimumLeaseRenewalInterval
+        {
+            get
+            {
+                return _minimumLeaseRenewalInterval;
+            }
+            set
+            {
+                _minimumLeaseRenewalInterval = value;
+            }
+        }
+
+        public async Task ReleaseLockAsync(IDistributedLock lockHandle, CancellationToken cancellationToken)
+        {
+            SingletonLockHandle singletonLockHandle = (SingletonLockHandle)lockHandle;
+
+            if (singletonLockHandle.LeaseRenewalTimer != null)
+            {
+                await singletonLockHandle.LeaseRenewalTimer.StopAsync(cancellationToken);
+            }
+
+            await ReleaseLeaseAsync(singletonLockHandle.Blob, singletonLockHandle.LeaseId, cancellationToken);
+        }
+
+        public async virtual Task<string> GetLockOwnerAsync(string account, string lockId, CancellationToken cancellationToken)
+        {
+            IStorageBlobDirectory lockDirectory = GetLockDirectory(account);
+            IStorageBlockBlob lockBlob = lockDirectory.GetBlockBlobReference(lockId);
+
+            await ReadLeaseBlobMetadata(lockBlob, cancellationToken);
+
+            // if the lease is Available, then there is no current owner
+            // (any existing owner value is the last owner that held the lease)
+            if (lockBlob.Properties.LeaseState == LeaseState.Available &&
+                lockBlob.Properties.LeaseStatus == LeaseStatus.Unlocked)
+            {
+                return null;
+            }
+
+            string owner = string.Empty;
+            lockBlob.Metadata.TryGetValue(FunctionInstanceMetadataKey, out owner);
+
+            return owner;
+        }
+
+        public async Task<IDistributedLock> TryLockAsync(
+            string account,
+            string lockId,
+            string lockOwnerId,
+            TimeSpan lockPeriod,
+            CancellationToken cancellationToken)
+        {
+            IStorageBlobDirectory lockDirectory = GetLockDirectory(account);
+            IStorageBlockBlob lockBlob = lockDirectory.GetBlockBlobReference(lockId);
+            string leaseId = await TryAcquireLeaseAsync(lockBlob, lockPeriod, cancellationToken);
+
+            if (string.IsNullOrEmpty(leaseId))
+            {
+                return null;
+            }
+
+            if (!string.IsNullOrEmpty(lockOwnerId))
+            {
+                await WriteLeaseBlobMetadata(lockBlob, leaseId, lockOwnerId, cancellationToken);
```

Comment:
Good catch. This is just inheriting the original implementation, so I don't think it's creating a new bug. But per your other comment, if we can get rid of owner altogether, this becomes moot.

---
In reply to: [120953083](https://github.com/Azure/azure-webjobs-sdk/pull/1184#discussion_r120953083) [](ancestors = 120953083)